### PR TITLE
Line channel updates

### PR DESCRIPTION
**Pull request by EricDahlvang**
Fixes https://github.com/Microsoft/BotBuilder/issues/5316
and
https://github.com/Microsoft/BotBuilder/issues/5313

OAuthPrompt (tested and works)

Dialogs Channel: added LIne 
     - SupportsSuggestedActions: 13
     - SupportsCardActions: 99

Channel Constant: added Line

Add ChoicesChannelTests

### CODE REVIEW

File: libraries/Microsoft.Bot.Builder.Dialogs/Choices/Channel.cs [hunk 1]
```diff
@@ -1,79 +1,85 @@
...
-        // This class has been deprecated in favor of the class in Microsoft.Bot.Connector.Channels located
-        // at https://github.com/Microsoft/botbuilder-dotnet/libraries/Microsoft.Bot.Connector/Channels.cs.
-        // This change is non-breaking and this class now inherits from the class in the connector library.
-        [Obsolete("This class is deprecated. Please use Microsoft.Bot.Connector.Channels.")]
-        public class Channels : Connector.Channels
-        {
-        }
-    }
-}
+using System;
```
Comment (cleemullins) [inline libraries/Microsoft.Bot.Builder.Dialogs/Choices/Channel.cs]: Do you know why this PR shows up as "Every line changed"?

File: libraries/Microsoft.Bot.Builder.Dialogs/Choices/Channel.cs [hunk 2]
```diff
@@ -1,79 +1,85 @@
...
+        public static bool SupportsSuggestedActions(string channelId, int buttonCnt = 100)
+        {
+            switch (channelId)
+            {
+                case Connector.Channels.Facebook:
+                case Connector.Channels.Skype:
+                    return buttonCnt <= 10;
+
+                case Connector.Channels.Line:
+                    return buttonCnt <= 13;
```
Comment (cleemullins) [inline libraries/Microsoft.Bot.Builder.Dialogs/Choices/Channel.cs]: Can you include a comment here pointing at the source of this information? As it sits here, I've no way of knowing "Is this right?". 

I know the other channels don't have this, but they're before my time and got grandfathered in.

File: libraries/Microsoft.Bot.Builder.Dialogs/Choices/Channel.cs [hunk 3]
```diff
@@ -1,79 +1,85 @@
...
+        public static bool SupportsCardActions(string channelId, int buttonCnt = 100)
+        {
+            switch (channelId)
+            {
+                case Connector.Channels.Facebook:
+                case Connector.Channels.Skype:
+                case Connector.Channels.Msteams:
+                    return buttonCnt <= 3;
+
+                case Connector.Channels.Line:
```
Comment (cleemullins) [inline libraries/Microsoft.Bot.Builder.Dialogs/Choices/Channel.cs]: Same comment - please include a link (or text) explaining where this value comes from.

File: libraries/Microsoft.Bot.Builder.Dialogs/Choices/Channel.cs [hunk 4]
```diff
@@ -1,79 +1,85 @@
...
-        public class Channels : Connector.Channels
-        {
-        }
-    }
-}
+using System;
+
+namespace Microsoft.Bot.Builder.Dialogs.Choices
+{
+    public class Channel
```
Comment (cleemullins) [inline libraries/Microsoft.Bot.Builder.Dialogs/Choices/Channel.cs]: As part of this PR, could you author XML Code comments for the methods?

File: tests/Microsoft.Bot.Builder.Dialogs.Tests/ChoicesChannelTests.cs [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+using Microsoft.Bot.Connector;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Bot.Builder.Dialogs.Tests
+{
+    [TestClass]
+    [TestCategory("Prompts")]
+    [TestCategory("Choice Tests")]
+    public class ChoicesChannelTests
```
Comment (cleemullins) [inline tests/Microsoft.Bot.Builder.Dialogs.Tests/ChoicesChannelTests.cs]: Love it!!

File: libraries/Microsoft.Bot.Builder.Dialogs/Choices/Channel.cs (hunk 1 above)
Comment (EricDahlvang) [reply libraries/Microsoft.Bot.Builder.Dialogs/Choices/Channel.cs]: I do not  :(

File: libraries/Microsoft.Bot.Builder.Dialogs/Choices/Channel.cs (hunk 2 above)
Comment (EricDahlvang) [reply libraries/Microsoft.Bot.Builder.Dialogs/Choices/Channel.cs]: Added as a comment: https://developers.line.biz/en/reference/messaging-api/#items-object

File: libraries/Microsoft.Bot.Builder.Dialogs/Choices/Channel.cs (hunk 3 above)
Comment (EricDahlvang) [reply libraries/Microsoft.Bot.Builder.Dialogs/Choices/Channel.cs]: I could find no docs stating Line supports 99 card actions, though I tried this on Line for Windows and Android.

File: tests/Microsoft.Bot.Builder.Dialogs.Tests/ChoicesChannelTests.cs (hunk 5 above)
Comment (EricDahlvang) [reply tests/Microsoft.Bot.Builder.Dialogs.Tests/ChoicesChannelTests.cs]: 👍

File: libraries/Microsoft.Bot.Builder.Dialogs/Choices/Channel.cs (hunk 4 above)
Comment (EricDahlvang) [reply libraries/Microsoft.Bot.Builder.Dialogs/Choices/Channel.cs]: Done

**Pull request merged**